src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+import duke.commands.Command;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+/**
+ * Main Duke class.
+ */
+public class Duke {
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+
+    public Duke() {
+        this(null);
+    }
+
+    public Duke(String directoryPath) {
+        this.ui = new Ui();
+        if (directoryPath != null) {
```

Comment:
Great structure of conditional statements and try catch blocks, very clear and concise for readers to understand!